src/Symfony/Component/Console/Command/Command.php

```diff
@@ -513,7 +513,12 @@ public function setHidden(bool $hidden /*= true*/)
      */
     public function isHidden()
     {
-        return $this->hidden;
+        $isHidden = false;
+        if (\PHP_VERSION_ID >= 80000 && $attribute = (new \ReflectionClass($this))->getAttributes(AsCommand::class)) {
+            $isHidden = $attribute[0]->newInstance()->hidden;
+        }
+
+        return $this->hidden || $isHidden;
```

Comment:
For reviewers (and for the author): this patch cannot be correct, because it makes the attribute a mandatory piece of configuration that cannot be ignored. This is against the root principles of attributes, which should be purely declarative.